src/UI/Component/Table/Factory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nhaageng@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * Table factory
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The Presentation Table lists some data from the system in a pleasant way.
+	 *       The user should be able to get a quick overview over records in a
+	 *       dataset as simple as possible; for this, only most relevant fields of
+	 *       a record are being displayed at first glance.
+	 *       Rows can be expanded to show more extensive fields and data
+	 *       less important for the identification of a record.
+	 *       The purpose of the presentation and exploration is known, and
+	 *       single records may only be explored in this way,
```

Comment:
Please lay a stronger emphasis here on: 
> this table aims at representing the dataset and not a list of single rows.

Maybe also elaborate a little the special role of records here. They seem to be different from entities. So what is their nature?